File: cpp/3021_flower_game/flower_game.cpp
#include <cassert>
#include <tuple>
#include <vector>

class Solution {
 public:
  long long flowerGame(int n, int m) { return (long long)m * n / 2; }
};

int main() {
  std::vector<std::tuple<int, int, long long>> tests{
      {3, 2, 3},
      {1, 1, 0},
  };

  for (auto &[n, m, expected] : tests) {
    assert(Solution().flowerGame(n, m) == expected);
  }
}

File: rust/3021_flower_game/flower_game.rs
struct Solution;

impl Solution {
    pub fn flower_game(n: i32, m: i32) -> i64 {
        (m as i64) * (n as i64) / 2
    }
}

fn main() {
    let tests = vec![(3, 2, 3), (1, 1, 0)];

    for (n, m, expected) in tests {
        assert_eq!(Solution::flower_game(n, m), expected);
    }
}
